modules/volume/src/processors/volumeregionmap.cpp

```diff
@@ -0,0 +1,116 @@
+
+namespace inviwo {
+
+// The Class Identifier has to be globally unique. Use a reverse DNS naming scheme
+const ProcessorInfo VolumeRegionMap::processorInfo_{
+    "org.inviwo.VolumeRegionMap",                               // Class identifier
+    "Volume Region Map",                                        // Display name
+    "Volume Operation",                                         // Category
+    CodeState::Stable,                                          // Code state
+    Tag::CPU | Tag{"Volume"} | Tag{"Atlas"} | Tag{"DataFrame"}  // Tags
+};
+const ProcessorInfo VolumeRegionMap::getProcessorInfo() const { return processorInfo_; }
+
+VolumeRegionMap::VolumeRegionMap()
+    : Processor()
+    , dataFrame_("mapping_indexes")
+    , inport_("inport")
+    , outport_("outport")
+    , from_{"from", "From", dataFrame_, ColumnOptionProperty::AddNoneOption::No, 1}
+    , to_{"to", "to", dataFrame_, ColumnOptionProperty::AddNoneOption::No, 2} {
+
+    addPort(inport_);
+    addPort(dataFrame_);
+    addPort(outport_);
+    addProperties(from_, to_);
+}
+
+namespace {
+constexpr auto copyColumn = [](const Column& col, auto& dstContainer, auto assign) {
+    col.getBuffer()->getRepresentation<BufferRAM>()->dispatch<void, dispatching::filter::Scalars>(
+        [&](auto buf) {
+            const auto& data = buf->getDataContainer();
+            for (auto&& [src, dst] : util::zip(data, dstContainer)) {
+                std::invoke(assign, src, dst);
+            }
+        });
+};
+}
+
+void VolumeRegionMap::process() {
+    // Get columns
+    auto csv = dataFrame_.getData();
+    auto oldIdx = csv->getColumn(from_.getSelectedValue());
+    auto newIdx = csv->getColumn(to_.getSelectedValue());
+    auto nrows = csv->getNumberOfRows();
+
+    // Store in vectors
+    std::vector<uint32_t> sourceIndices(nrows);
+    std::vector<uint32_t> destinationIndices(nrows);
+
+    copyColumn(*oldIdx, sourceIndices,
+               [](const auto& src, auto& dst) { dst = static_cast<uint32_t>(src); });
+    copyColumn(*newIdx, destinationIndices,
+               [](const auto& src, auto& dst) { dst = static_cast<uint32_t>(src); });
+
+    // Volume
+    auto inVolume = inport_.getData();
+    auto newVolume = std::shared_ptr<Volume>(inVolume->clone());
+    auto volRep = newVolume->getEditableRepresentation<VolumeRAM>();
+
+    volRep->dispatch<void, dispatching::filter::Scalars>([&](auto volram) {
+        using ValueType = util::PrecisionValueType<decltype(volram)>;
+        ValueType* dataPtr = volram->getDataTyped();
+
+        const auto& dim = volram->getDimensions();
+
+        std::transform(dataPtr, dataPtr + dim.x * dim.y * dim.z, dataPtr, [&](const ValueType& v) {
+            for (size_t i = 0; i < sourceIndices.size(); ++i) {
+                if (static_cast<uint32_t>(v) == sourceIndices[i]) {
+                    return static_cast<ValueType>(destinationIndices[i]);
+                }
+            }
+            return ValueType{0};
```

Comment:
Perhaps this should be a user-specifed value that defaults to 0?

Reply:
There is now an option to keep values that are missing, or set a specific value. Currently, it will default to 0 unless changed in GUI.